extensions/src/ACRE2Steam/CallExt_DllMain.cpp

```diff
@@ -287,249 +59,178 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function) {
-                    for (auto unique_ts : unique_ts_locations) {
-                        if (location == unique_ts) {
-                            found = true;
-                            break;
-                        }
-                    }
-                    if (!found) {
-                        unique_ts_locations.push_back(location);
+            bool try_copy = false;
 
-                        // If we are going to copy to a folder do not remove from the plugin from it.
-                        unique_delete_ts_locations.erase(std::remove(unique_delete_ts_locations.begin(), unique_delete_ts_locations.end(), location), unique_delete_ts_locations.end());
-                    }
+            do {
+                std::future<idi::acre::UpdateCode> update_ts3 =
+                  std::async(std::launch::async, [&]() { return ts3_plugin.handle_update_plugin(); });
+                std::future<idi::acre::UpdateCode> update_mumble =
+                  std::async(std::launch::async, [&]() { return mumble_plugin.handle_update_plugin(); });
+
+                const idi::acre::UpdateCode ts3_update_result    = update_ts3.get();
+                const idi::acre::UpdateCode mumble_update_result = update_mumble.get();
+
+                std::string error_msg;
+                const bool update_ts3_ok =
+                  (ts3_update_result != idi::acre::UpdateCode::update_failed) && (ts3_update_result != idi::acre::UpdateCode::other);
+                if (!update_ts3_ok) {
+                    error_msg = ts3_plugin.get_last_error_message();
                 }
-            }
 
-            // No locations to copy to.
-            if (unique_ts_locations.size() == 0) {
-                const int32_t result = MessageBoxA(nullptr, "ACRE2 was unable to find a TeamSpeak 3 installation. If you do have an installation please copy the plugins yourself or reinstall TeamSpeak 3. \n\n If you are sure you have TeamSpeak 3 installed and wish to prevent this message from appearing again remove ACRE2Steam.dll and ACRE2Steam_x64.dll from your @acre2 folder.\n\nContinue anyway?", "ACRE2 Installation Error", MB_YESNO | MB_ICONEXCLAMATION);
-                if (result == IDYES) {
-                    strncpy(output, "[-3,true]", outputSize);
-                    return;
+                const bool update_mumble_ok =
+                  (mumble_update_result != idi::acre::UpdateCode::update_failed) && (mumble_update_result != idi::acre::UpdateCode::other);
+                if (!update_mumble_ok) {
+                    error_msg = mumble_plugin.get_last_error_message();
+                }
+
+                if (!update_ts3_ok || !update_mumble_ok) {
+                    std::ostringstream oss;
+                    oss << "ACRE2 was unable to copy the Mumble/TeamSpeak 3 plugin. Please check if you have write access to the plugin "
+                        << "folder, close any instances of Mumble and/or TeamSpeak 3 and/or Mumble and click \"Try Again\".\n\nIf you "
```

Comment:
```suggestion
                        << "folder, close any instances of TeamSpeak 3 and/or Mumble and click \"Try Again\".\n\nIf you "
```
Mumble is mentioned twice `Mumble and/or TeamSpeak 3 and/or Mumble`